parcels/particleset.py

```diff
@@ -382,6 +382,45 @@ def show(self, **kwargs):
             print('Plot saved to '+savefile+'.png')
             plt.close()
 
+    def density(self, field=None, particle_val=None, relative=False, area_scale=True):
+        lons = [p.lon for p in self.particles]
+        lats = [p.lat for p in self.particles]
+        if field is not None:
+            # Kick out particles that are not within the limits of our density field
+            half_lon = (field.lon[1] - field.lon[0])/2
+            half_lat = (field.lat[1] - field.lat[0])/2
+            dparticles = (lons > (np.min(field.lon)-half_lon)) * (lons < (np.max(field.lon)+half_lon)) * \
+                         (lats > (np.min(field.lat)-half_lat)) * (lats < (np.max(field.lat)+half_lat))
+            dparticles = np.where(dparticles)[0]
+        else:
+            field = self.grid.U
+            dparticles = range(len(self.particles))
+        Density = np.zeros((field.lon.size, field.lat.size), dtype=np.float32)
+
+        # For each particle, find closest vertex in x and y and add 1 or val to the count
+        if particle_val is not None:
+            for p in dparticles:
+                Density[np.argmin(np.abs(lons[p] - field.lon)), np.argmin(np.abs(lats[p] - field.lat))] \
+                    += getattr(self.particles[p], particle_val)
+        else:
+            for p in dparticles:
+                nearest_lon = np.argmin(np.abs(lons[p] - field.lon))
+                nearest_lat = np.argmin(np.abs(lats[p] - field.lat))
+                Density[nearest_lon, nearest_lat] += 1
+            if relative:
+                Density /= len(dparticles)
+
+        if area_scale:
+            area = np.zeros(np.shape(field.data[0, :, :]), dtype=np.float32)
+            dx = (field.lon[1] - field.lon[0]) * 1852 * 60 * np.cos(field.lat*np.pi/180)
+            dy = (field.lat[1] - field.lat[0]) * 1852 * 60
```

Comment:
Ok, while we may not be able to internalise this conversion directly, it would be great if it were done by one of our internal `Converter` classes. For that I would probably create a `geoconv = Geographic()` or a `polarconv=GeographicPolar()` at the top of this routine, or better yet pull the converter from the field, and call `dx = polarconv.to_source(field.lon[1] - field.lon[0])`. However, as I'm writing this I'm realising that I haven't implemented the inverse call `to_source()` yet, so I'm happy for this to go in as is and be fixed later.